ansible/roles/node_exporter_nginx/templates/node_exporter_nginx.conf.j2

```diff
@@ -0,0 +1,28 @@
+server {
+    server_name _;
+
+    listen 9100 ssl default_server;
+    listen [::]:9100 ssl default_server;
+
+    ssl_certificate {{ node_exporter_ssl_cert_path }};
+    ssl_certificate_key {{ node_exporter_ssl_key_path }};
+    ssl_trusted_certificate {{ node_exporter_ssl_cert_path }};
+    # Recommendations from https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
+    ssl_protocols TLSv1.1 TLSv1.2;
+    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
+    ssl_prefer_server_ciphers on;
+    ssl_session_cache shared:SSL:10m;
+
+  location / {
+    auth_basic "Monitoring realm";
+    auth_basic_user_file /etc/nginx/prometheus.htpasswd;
+
+    proxy_pass http://{{ node_exporter_listen_address }};
+    proxy_http_version 1.1;
+    proxy_set_header Host $http_host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto $scheme;
```

Comment:
`node_exporter` doesn't care about those.  The only thing it really wants to know is the encoding headers so it can correctly respond with text or protobuf.  Also compression encoding.